### PR TITLE
Feat--update-transmuter-oracle-+-rebalancing-ERNX

### PR DESCRIPTION
**Pull request by GuillaumeNervoXS**

### CODE REVIEW

File: helpers/createTx.sh [hunk 1]
```diff
@@ -104,16 +106,15 @@ function main {
...
-        echo ""
-        echo "Would you like to execute the script ? (yes/no)"
-        read execute
-
-        if [[ $execute == "yes" ]]; then
-            yarn submit:foundry
-        fi
+        # if [[ $execute == "yes" ]]; then
+        #     yarn submit:foundry
+        # fi
```
Comment (0xtekgrinder) [inline helpers/createTx.sh:117]: maybe revert these changes to have a clean script file ?

File: scripts/foundry/Utils.s.sol [hunk 2]
```diff
@@ -37,29 +38,31 @@ contract Utils is Script, CommonUtils {
...
-        forkIdentifier[CHAIN_LINEA] = lineaFork;
+        // forkIdentifier[CHAIN_OPTIMISM] = optimismFork;
+        // forkIdentifier[CHAIN_POLYGON] = polygonFork;
+        // forkIdentifier[CHAIN_GNOSIS] = gnosisFork;
+        // forkIdentifier[CHAIN_BNB] = bnbFork;
+        // forkIdentifier[CHAIN_CELO] = celoFork;
+        // forkIdentifier[CHAIN_POLYGONZKEVM] = polygonZkEVMFork;
+        // forkIdentifier[CHAIN_BASE] = baseFork;
+        // forkIdentifier[CHAIN_LINEA] = lineaFork;
+        forkIdentifier[CHAIN_FORK] = localFork;
```
Comment (0xtekgrinder) [inline scripts/foundry/Utils.s.sol:65]: revert the fork identifiers to have a default implementation (maybe in the future I will create something more complex to create the fork only if the env is there)

File: test/transmuter/TransmuterUpdateFacetsTest.t.sol [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+    //
+
+    function setUp() public override(Utils, BaseTest) {
+        BaseTest.setUp();
+
+        // special case as we rely on the fork state
+        vm.selectFork(forkIdentifier[CHAIN_FORK]);
+        // vm.selectFork(forkIdentifier[chainId]);
+
+        vm.pauseGasMetering();
```
Comment (0xtekgrinder) [inline test/transmuter/TransmuterUpdateFacetsTest.t.sol:36]: why did you needed to use this ?

File: test/transmuter/TransmuterUpdateFacetsTest.t.sol [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+        );
+
+        uint256 chainId = json.readUint("$.chainId");
+
+        address gnosisSafe = _chainToContract(chainId, ContractType.GovernorMultisig);
+
+        transmuter = ITransmuter(_chainToContract(chainId, ContractType.TransmuterAgEUR));
+
+        address to = json.readAddress("$.to");
+        // uint256 value = json.readUint("$.value");
```
Comment (0xtekgrinder) [inline test/transmuter/TransmuterUpdateFacetsTest.t.sol:53]: it is not real usefull but it should be a best practice to keep the value and send it to the gnosisSafe.call() as we never know if we wants to change the script someday

File: test/transmuter/TransmuterUpdateFacetsTest.t.sol [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+        MockSafe mockSafe = new MockSafe();
+        vm.etch(gnosisSafe, address(mockSafe).code);
+        vm.prank(gnosisSafe);
+        (bool success, ) = gnosisSafe.call(abi.encode(address(to), payload, operation, 1e6));
+        if (!success) revert();
+    }
+
+    function test_script() external {
+        // Now test that everything is as expected
+        uint256 chainId = CHAIN_SOURCE;
```
Comment (0xtekgrinder) [inline test/transmuter/TransmuterUpdateFacetsTest.t.sol:67]: should be able to use chainId from transaction.json instead of CHAIN_SOURCE

File: test/transmuter/TransmuterUpdateFacetsTest.t.sol (hunk 3 above)
Comment (GuillaumeNervoXS) [reply test/transmuter/TransmuterUpdateFacetsTest.t.sol:38]: revert (wrong gas estimation I guess)

**Review by sogipec — APPROVED** [review]
